modules/extension/wmts/src/main/java/org/geotools/ows/wmts/map/WMTSCoverageReader.java

```diff
@@ -326,11 +328,22 @@ GridCoverage2D getMap(ReferencedEnvelope requestedEnvelope, int width, int heigh
             renderTiles(responses, image.createGraphics(), requestedEnvelope, targetTransform);
 
             return gcf.create(layer.getTitle(), image, global);
-        } catch (ServiceException e) {
+        } catch (ServiceException | FactoryException | TransformException e) {
             throw new IOException("GetMap failed", e);
         }
     }
 
+    private ReferencedEnvelope toEastNorthAxisOrder(ReferencedEnvelope extent)
```

Comment:
We really need a comment here explain what this method does and why it does it - otherwise it looks very suspicious.